src/hash.rs

```diff
@@ -59,8 +60,17 @@ impl StarkFelt {
         Err(StarknetApiError::OutOfRange { string: hex_str_from_bytes::<32, true>(bytes) })
     }
 
+    /// Returns a new unckecked [`StarkFelt`].
+    ///
+    /// # Safety
+    /// The most significant byte nible of the array must be 0 (`byte[0] < 0x10`).
+    /// Otherwise the resulting [`StarkFelt`] will end up containing an out of range value.
+    pub const unsafe fn new_unchecked(bytes: [u8; 32]) -> Self {
```

Comment:
Should this function visibility be restrained to pub(crate)?

I don't think it's necessary. The unsafe make users aware of what they are doing.

Reply:
it's not necessary